autonomi/src/client/data_types/pointer.rs

```diff
@@ -71,60 +109,167 @@ impl Client {
             })?;
 
         // verify payment was successful
-        let (proof, _price) = match payment_proofs.get(&xor_name) {
-            Some((proof, price)) => (proof, price),
+        let (proof, price) = match payment_proofs.get(&xor_name) {
+            Some((proof, price)) => (Some(proof), price),
             None => {
-                error!("Pointer at address: {address:?} was already paid for");
-                return Err(PointerError::PointerAlreadyExists(address));
+                info!("Pointer at address: {address:?} was already paid for, update is free");
```

Comment:
need to cross check with the design that whether an update to the PointData is free or shall be paid.
if need to `pay for update` , the `quoting` flow need to be double checked as well.